### PR TITLE
Feature/cell editable custom component

### LINKED ISSUE #2559: How to use custom edit component when using cell editable feature ?
I am using Material-Table in my project. I wanted to use a custom edit component when using **cell editable feature**, as I didn't want to use rowUpdate. I tried passing the custom component using "editComponent" prop in columns, but it seems to have no effect.

Here is the codesandbox link of an example implementation - [https://codesandbox.io/s/material-demo-forked-vp145](https://codesandbox.io/s/material-demo-forked-vp145)

If this cannot be done, please consider this as a feature request.

### PR DESCRIPTION
**Pull request by klassm**
## Related Issue

Resolves #2559 - using a custom component for cell editables.

## Description

Before this change setting an editComponent for cellEditable tables had no effect. Now we just render the component a user passes to us.

## Impacted Areas in Application

* cell editables

## Additional Notes

Thank you so much for this project. It really makes life much easier when it comes to displaying table like data. Especially the tree view is so useful. Keep up the great work :-)****

### CODE REVIEW

**Comment by klassm** [comment]
Eslint now added a lot of changes. Should I revert those?

**Comment by matthewoestreich** [comment]
@klassm thank you for this! It looks like your IDE is changing single quotes to double quotes. I have updated the .eslintignore file.

Would you mind trying once more?

File: .github/README_ORIGINAL.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ A simple and powerful Datatable for React based on [Material-UI Table](https://m
 [![Build Status](https://travis-ci.org/mbrn/material-table.svg?branch=master)](https://travis-ci.org/mbrn/material-table)
 [![Financial Contributors on Open Collective](https://opencollective.com/material-table/all/badge.svg?label=financial+contributors)](https://opencollective.com/material-table) [![npm package](https://img.shields.io/npm/v/material-table/latest.svg)](https://www.npmjs.com/package/material-table)
 [![NPM Downloads](https://img.shields.io/npm/dm/material-table.svg?style=flat)](https://npmcharts.com/compare/material-table?minimal=true)
-[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/mbrn/material-table.svg)](http://isitmaintained.com/project/mbrn/material-table "Average time to resolve an issue")
+[![Average time to resolve an issue](http://isitmaintained.com/badge/resolution/mbrn/material-table.svg)](http://isitmaintained.com/project/mbrn/material-table 'Average time to resolve an issue')
```
Comment (matthewoestreich) [inline .github/README_ORIGINAL.md:19]: Should be single quotes.

File: .github/README_ORIGINAL.md [hunk 2]
```diff
@@ -108,14 +108,14 @@ To install @material-ui/icons with `yarn`:
 If your environment doesn't support tree-shaking, the **recommended** way to import the icons is the following:
 
 ```jsx
-import AddBox from "@material-ui/icons/AddBox";
-import ArrowDownward from "@material-ui/icons/ArrowDownward";
+import AddBox from '@material-ui/icons/AddBox';
```
Comment (matthewoestreich) [inline .github/README_ORIGINAL.md:111]: Should be single quotes

File: .github/README_ORIGINAL.md [hunk 3]
```diff
@@ -172,29 +172,29 @@ const tableIcons = {
 Here is a basic example of using material-table within a react application.
 
 ```jsx
-import React, { Component } from "react";
-import ReactDOM from "react-dom";
-import MaterialTable from "material-table";
+import React, { Component } from 'react';
+import ReactDOM from 'react-dom';
+import MaterialTable from 'material-table';
```
Comment (matthewoestreich) [inline .github/README_ORIGINAL.md:177]: Should be single quotes

File: material-table.iml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (matthewoestreich) [inline material-table.iml]: This file needs to be removed as well. I added .iml to .gitignore

**Comment by klassm** [comment]
@oze4 Thanks for the review :-) I've done the merge one more time, this time from your latest changes in master. It looks like everything is now single quotes.

**Review by matthewoestreich — APPROVED** [review]

**Review by matthewoestreich — APPROVED** [review]

**Comment by klassm** [comment]
Thanks :-)

**Comment by matthewoestreich** [comment]
@klassm v2.2.0 has been published with these changes. Thank you so much for contributing!

**Comment by ItzikHendelKahun** [comment]
@oze4, I need this fix in my project but it seems the last published release is 1.69.2 which not includes this fix

**Comment by matthewoestreich** [comment]
The latest version is 2.3.29. It sounds like you are looking at mbrn/material-table

**Comment by ItzikHendelKahun** [comment]
@oze4 , Thanks a lot. I did not notice that it's a forked branch from mbrn/material-table. Do you plan to merge into mbrn/material-table at some point?

**Comment by gas6262** [comment]
Where is the documentation for this feature?

**Comment by matthewoestreich** [comment]
For what feature? 

I don't think we are against merging upstream.

**Pull request merged**